examples/joy_circle_grid.ml

```diff
@@ -0,0 +1,34 @@
+open Joy.Shape
+
+let grid_of_circles canvas_width canvas_height spacing =
+  let create_circle x y = circle 50 |> translate x y in
+  let num_circles_x = canvas_width / spacing in
+  let num_circles_y = canvas_height / spacing in
+  let initial_x = (canvas_width - (num_circles_x * spacing)) / 2 in
```

Comment:
Can you explain this calculation please?

Reply:
the `create_circle` function creates a circle of radius 50. It takes x and y parameters and moves the circle to the specified coordinates
`num_circles_x` and `num_circles_y` are variables that determine how many circles can fit horizontally and vertically respectively into the canvas
`initial_x` calculates the initial x coordinate for the first circle in the grid

Reply:
How is `initial_x` calculated?